src/Delegate/NotFoundDelegate.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-expressive for the canonical source repository
+ * @copyright Copyright (c) 2017 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-expressive/blob/master/LICENSE.md New BSD License
+ */
+
+namespace Zend\Expressive\Delegate;
+
+use Fig\Http\Message\StatusCodeInterface;
+use Interop\Http\ServerMiddleware\DelegateInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Zend\Expressive\Template\TemplateRendererInterface;
+
+class NotFoundDelegate implements DelegateInterface
+{
+    const TEMPLATE_DEFAULT = 'error::404';
+
+    /**
+     * @var TemplateRendererInterface
+     */
+    private $renderer;
+
+    /**
+     * This duplicates the property in StratigilityNotFoundHandler, but is done
+     * to ensure that we have access to the value in the methods we override.
+     *
+     * @var ResponseInterface
+     */
+    protected $responsePrototype;
+
+    /**
+     * @var string
+     */
+    private $template;
+
+    /**
+     * @param ResponseInterface $responsePrototype
+     * @param TemplateRendererInterface $renderer
+     * @param string $template
+     */
+    public function __construct(
+        ResponseInterface $responsePrototype,
+        TemplateRendererInterface $renderer = null,
+        $template = self::TEMPLATE_DEFAULT
+    ) {
+        $this->responsePrototype = $responsePrototype;
+        $this->renderer          = $renderer;
+        $this->template          = $template;
+    }
+
+    /**
+     * Creates and returns a 404 response.
+     *
+     * @param ServerRequestInterface $request  Ignored.
```

Comment:
Only one space before param description please.
And not sure if the description is clear enough? Is `$request` ignored? I can see it pass thru, so maybe it's not (?)